api/src/main/java/org/openmrs/module/bedmanagement/service/impl/BedManagementServiceImpl.java

```diff
@@ -405,8 +405,6 @@ public BedType retireBedType(BedType bedType, String retireReason) {
 		}
 		bedType.setRetired(true);
 		bedType.setRetireReason(retireReason);
-		bedType.setRetiredBy(Context.getAuthenticatedUser());
-		bedType.setDateRetired(new Date());
```

Comment:
Did you confirm that this actually works without saving via the service layer with Context.getService()?